pkg/queue/bounded_queue.go

```diff
@@ -164,6 +166,8 @@ func (q *BoundedQueue) Resize(capacity int) bool {
 	// #nosec
 	swapped := atomic.CompareAndSwapPointer((*unsafe.Pointer)(unsafe.Pointer(&q.items)), unsafe.Pointer(q.items), unsafe.Pointer(&queue))
 	if swapped {
+		atomic.CompareAndSwapInt32(&q.capacity, int32(q.Capacity()), int32(capacity))
```

Comment:
queue capacity is determined by the channel in L47, updating this field does not achieve anything.

Reply:
True. I had in mind the case where two channels exist at a given time after a resize operation, but the new channel will indeed have the right capacity value if the previous pointer swap happened successfully.